include/spblas/views/matrix_opt_impl.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <spblas/backend/concepts.hpp>
+#include <spblas/backend/cpos.hpp>
+#include <spblas/concepts.hpp>
+
+#ifdef SPBLAS_ENABLE_ONEMKL_SYCL
+#include <oneapi/mkl.hpp>
+#include <sycl/sycl.hpp>
+#endif
+
+namespace spblas {
+
+template <matrix M>
+  requires(view<M> &&
+           (__detail::is_csr_view_v<M> || __detail::is_csc_view_v<M>) )
+class matrix_opt<M> : public view_base {
+public:
+  using scalar_type = tensor_scalar_t<M>;
+  using scalar_reference = tensor_scalar_reference_t<M>;
+  using index_type = tensor_index_t<M>;
+  using offset_type = tensor_offset_t<M>;
+
+  matrix_opt(M matrix) : matrix_(matrix) {
+#ifdef SPBLAS_ENABLE_ONEMKL_SYCL
+    matrix_handle_ = nullptr;
+#endif
+  }
+
+  ~matrix_opt() {
+#ifdef SPBLAS_ENABLE_ONEMKL_SYCL
+    if (matrix_handle_) {
+      // q here needs to be on same context as queue in operations,
+      // idealy from execution policy
+      sycl::queue q(sycl::cpu_selector_v);
+      oneapi::mkl::sparse::release_matrix_handle(q, &matrix_handle_, {}).wait();
+      matrix_handle_ = nullptr;
+    }
+#endif
+  }
+
+  auto shape() const noexcept {
+    return __backend::shape(base());
+  }
+
+  index_type size() const noexcept {
+    return __backend::size(base());
+  }
+
+  auto base() {
+    return matrix_;
+  }
+
+  auto base() const {
+    return matrix_;
+  }
```

Comment:
why do we need both const (accessor) and nonconst(mutator) functions for base() ?  can we really guarantee that the accessor won't modify matrix_ ?